boot/zephyr/Kconfig

```diff
@@ -28,6 +27,19 @@ partition=MCUBOOT_PAD
 partition-size=0x200
 source "${ZEPHYR_BASE}/../nrf/subsys/partition_manager/Kconfig.template.partition_size"
 
+config PM_PARTITION_SIZE_MCUBOOT
+	hex "Flash space reserved for the MCUboot image"
+	default 0xc000 if !BOOT_MIN_PARTITION_SIZE
```

Comment:
I'm not sure having a `BOOT_MIN_PARTITION_SIZE` setting is the best thing here.

This pattern will start to give n-kconfig symbols, one for each `minimal`, `external`, `mycustom`, `...` variant.

It's is already possible to do:
```
CONFIG_PM_PARTITION_SIZE_MCUBOOT=0x7c00
```
in the `prj_minimal.conf`

And then just keep `0xc000` as default if nothing is specified.

Reply:
The idea here was to align the partition image sizing scheme here with [b0's scheme](https://github.com/nrfconnect/sdk-nrf/blob/master/subsys/bootloader/Kconfig#L162), which uses this `<bootloader>_MIN_PARTITION_SIZE` define.

> It's is already possible to do:
>
> CONFIG_PM_PARTITION_SIZE_MCUBOOT=0x7c00
>
> in the prj_minimal.conf
>
> And then just keep 0xc000 as default if nothing is specified.

Yes but this will automatically handle sizing for general, nRF5340, nRF9160, and external crypto, without further input from the user. Isn't this streamlining a good thing for ease of use and future-proofing?

Reply:
> Isn't this streamlining a good thing for ease of use and future-proofing?

Streamlining is always a good thing, but adding yet another symbols that users must consider whether it should be true or false doesn't necessarily make it easier to understand the system, especially not when there is already a setting for handling the size .

Secondly, setting `CONFIG_BOOT_MIN_PARTITION_SIZE=y` doesn't mean this:
`Build for a minimal image size`, it means, set the default value of `PM_PARTITION_SIZE_MCUBOOT` to a minimal size.

It's still the actual value of `PM_PARTITION_SIZE_MCUBOOT` that decides the size, and not `CONFIG_BOOT_MIN_PARTITION_SIZE=y` so the name and text for the symbol is misleading.

> without further input from the user. 

User still needs to build with: `-DCONF_FILE=prj_minimal.conf` so I don't see how having `CONFIG_PM_PARTITION_SIZE_MCUBOOT=0x7c00` instead of `CONFIG_BOOT_MIN_PARTITION_SIZE=y` is easier for the user.
Could you elaborate a bit on the use-case you have in mind where the proposed change is actually making it easier fir the user ? Maybe that could even lead to an improved outcome.

Reply:
@hakonfam the Kconfig symbol that @kyberdin refers to in this commit: https://github.com/nrfconnect/sdk-nrf/commit/3ad0ab5c25b5b466f0939094fb26ef85e08cd1c3 is actually poorly named, as it doesn't ensure building for minimal size.

It only sets a different default value, it doesn't ensure that minimum size is used.
```
config B0_MIN_PARTITION_SIZE
	bool "Use minimimum partition size"

....
config PM_PARTITION_SIZE_B0_IMAGE
	....
	default 0x8000 if !B0_MIN_PARTITION_SIZE
	....
	default 0x7000
```

Reply:
@tejlmand Sorry for taking so long to get back to this. After more thought, I agree that `CONFIG_BOOT_MIN_PARTITION_SIZE=y` is misleading vs specifying via `CONFIG_PM_PARTITION_SIZE_MCUBOOT`. Setting a partition value this way in the `prj_minimal.conf` or `external_crypto.conf` is straightforward and makes sense.

However, a bit of a dilemma by removing this:

Let's say:
* Default partition size is `0xc000` in Kconfig.
* `prj_minimal` sets the partition size to `0x7c00`.
* `external_crypto.conf` sets partition size to `0x3e00`

nRF9160 flash blocks for MCUBoot should be in chunks of at least `FPROTECT_BLOCK_SIZE`, but only Kconfig can set it to this define. A `boards/nrf9160_nrf9160_minimal.conf` could set the partition size to a hardcoded value matching `FPROTECT_BLOCK_SIZE`, but `external_crypto.conf` will override everything regardless.

Considering that this PR aims to enable users to:

1. Build a minimal mcuboot
2. Minimize it further with external crypto

I don't see a way of successfully accomplishing both of these goals for current and future device needs by relying on a hardcoded value in the `external_crypto.conf` fragment.

Therefore, I think the current use case works because:

* It handles the necessary logic across boards
* Defining `CONFIG_PM_PARTITION_SIZE_MCUBOOT=xxxx` is always available to a user fragment for overriding everything, should they want to
* Since `CONFIG_BOOT_MIN_PARTITION_SIZE=y` is set in `prj_minimal.conf`, the user doesn't need to remember or interact with it when building a minimal mcuboot with `-Dmcuboot_CONF_FILE=<path to>/prj_minimal.conf`.

To better clarify the Kconfig option for this, perhaps we could rename `BOOT_MIN_PARTITION_SIZE` to `BOOT_USE_MIN_PARTITION_SIZE`?

Similarly, the Kconfig could be refactored into something like (while trying to avoid ["stuck" configs](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig)):

```
config BOOT_USE_MIN_PARTITION_SIZE
	bool "Produce an MCUboot image that has a minimal partition size."
        help
          Enable this to build MCUboot with the minimum partition size for its 

config PM_PARTITION_SIZE_MCUBOOT
	default FPROTECT_BLOCK_SIZE if (SOC_NRF5340_CPUAPP || SOC_NRF9160)
	default 0x3e00 if BOOT_NRF_EXTERNAL_CRYPTO
	default 0x7c00
	depends on BOOT_MIN_PARTITION_SIZE

config PM_PARTITION_SIZE_MCUBOOT
	hex "Flash space allocated for the MCUboot partition." if !BOOT_USE_MIN_PARTITION_SIZE
	default 0xc000
	help
	  Flash space set aside for the MCUboot partition.
```

Reply:
>  Similarly, the Kconfig could be refactored into something like (while trying to avoid "stuck" configs)

As long as you have a prompt, then you will hit the stuck configs.
https://github.com/kyberdin/sdk-mcuboot/blob/eb69274466d280aedc724a3e6e0f55d1fab56717/boot/zephyr/Kconfig#L43

Reply:
@tejlmand I have replaced `BOOT_MIN_PARTITION_SIZE` with `BOOT_USE_MIN_PARTITION_SIZE` for the clarity mentioned in my previous comment.

Adding `depends on` to the partition prompt failed to build due to extra tokens, so I used `if !BOOT_USE_MIN_PARTITION_SIZE` in its place.